impl-tinkerpop/src/main/java/org/hawkular/inventory/impl/tinkerpop/AbstractSourcedGraphService.java

```diff
@@ -78,6 +93,121 @@ public Single create(Blueprint blueprint) {
         return createSingleBrowser(FilterApplicator.fromPath(path).get());
     }
 
+    public final void update(E entity) {
+        Constants.Type type = Constants.Type.of(entity);
+        List<String> mappedProperties = Arrays.asList(type.getMappedProperties());
+
+        entity.getProperties().keySet().forEach(k -> {
+            if (mappedProperties.contains(k)) {
+                throw new IllegalArgumentException("Property '" + k + "' is reserved. Cannot set it to a custom value");
+            }
+        });
+
+        Vertex vertex = convert(entity);
```

Comment:
`convert(entity)` might an expensive op (assuming we don't have any global index) what about using 

```java
        HawkularPipeline<?, Vertex> vertices = source().hasUid(entity.getId());
        if (!vertices.hasNext()) throw ENFE..
        Vertex vertex = vertices.next();
```